Description: tune flags to build libv8 on armel
 * mno-thumb-interwork because v8 supports thumb-interwork on armv5t.
   On debian, armel arch targets armv4t, with softfp, so we use that to know where to put the flag.
   armhf arch, where thumb-interwork is supported, uses hardfloat.
 * fno-tree-sink to workaround a gcc armel bug
 * Wno-psabi to fix gcc warnings on armel
Forwarded: not-needed
Author: Antonio Radici <antonio@dyne.org>
Author: Jérémy Lal <kapouer@melix.org>
Bug-V8: http://code.google.com/p/v8/issues/detail?id=590
Bug-GCC: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=39604
Bug-GCC: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=42748
Last-Update: 2010-11-25
--- a/SConstruct
+++ b/SConstruct
@@ -149,19 +149,19 @@
       'armeabi:soft' : {
         'CPPDEFINES' : ['USE_EABI_HARDFLOAT=0'],
         'simulator:none': {
-          'CCFLAGS':     ['-mfloat-abi=soft'],
+          'CCFLAGS':     ['-mfloat-abi=soft', '-Wno-psabi', '-fno-tree-sink'],
         }
       },
       'armeabi:softfp' : {
         'CPPDEFINES' : ['USE_EABI_HARDFLOAT=0', 'CAN_USE_VFP_INSTRUCTIONS'],
         'simulator:none': {
-          'CCFLAGS':     ['-mfloat-abi=softfp'],
+          'CCFLAGS':     ['-mfloat-abi=softfp', '-mno-thumb-interwork', '-Wno-psabi', '-fno-tree-sink'],
         }
       },
       'armeabi:hard' : {
         'CPPDEFINES' : ['USE_EABI_HARDFLOAT=1', 'CAN_USE_VFP_INSTRUCTIONS'],
         'simulator:none': {
-          'CCFLAGS':     ['-mfloat-abi=hard'],
+          'CCFLAGS':     ['-mfloat-abi=hard', '-Wno-psabi', '-fno-tree-sink'],
         }
       }
     },
